src/utils/proxyRequest/provider/ServerProvider.js

```diff
@@ -0,0 +1,89 @@
+import { http } from '@/utils';
+
+export default class ServerProvider {
+  constructor(serverUrl) {
+    this.url = serverUrl;
+  }
+
+  add(params) {
+    return this.write(params);
+  }
+
+  readProp = async params => {
+    try {
+      const { prop } = params;
+      const data = await this.read(params);
+
+      return data[prop];
+    } catch (e) {
+      e.title = 'Error in server storage';
+      e.text = "Can't read data from server storage, maybe it is not available";
+      e.type = 'is-warning';
+
+      throw e;
+    }
+  };
+
+  read = async params => {
+    try {
+      const { url } = params;
+      const { data } = await http.get(url);
+
+      return data;
+    } catch (e) {
+      e.title = 'Error in server storage';
+      e.text = "Can't read data from server storage, maybe it is not available";
+      e.type = 'is-warning';
+
+      throw e;
+    }
+  };
+
+  writeProp = async params => {
+    try {
+      const { url, payload, prop } = params;
+      const { data } = await this.write(url, { [prop]: payload });
+
+      return data;
+    } catch (e) {
+      e.title = 'Error in server storage';
+      e.text = "Can't save data to server storage, maybe it is not available";
+      e.type = 'is-warning';
+
+      throw e;
+    }
+  };
+
+  write = async params => {
+    try {
+      const { url, payload } = params;
+      const { data } = await http.post(url, payload);
+
+      return data;
+    } catch (e) {
+      e.title = 'Error in server storage';
+      e.text = "Can't save data to server storage, maybe it is not available";
+      e.type = 'is-warning';
+
+      throw e;
+    }
+  };
+
+  remove = async params => {
+    try {
+      const { url, payload } = params;
+      const { data } = await http.delete(url, payload);
+
+      return data;
+    } catch (e) {
+      e.title = 'Error in server storage';
+      e.text =
+        "Can't remove data from server storage, maybe it is not available";
+      e.type = 'is-warning';
+
```

Comment:
[The same story.](https://github.com/endpass/endpass-wallet/pull/194/files#diff-44a2349c1ec279a1174ffb82d37490faR22)